### PR TITLE
added flan-t5 example into test

### PR DESCRIPTION
**Pull request by bm777**
Hey @EricLBuehler, I didn't find time to work on creating a pipeline for t5. 
Since I'm full-time working on my startup project, however, I do succeed in having` flan-t5 `running in my hybrid system. 
As promised, please find this `flan-t5` impl in the test/ folder.

----
I do sync the cargo.toml file, try to run the `test_flan-t5-quantized.rs` file.
And this PR is on top of your latest commit.

### CODE REVIEW

File: tests/test_flan-t5-quantized.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    which: Which,
+}
+
+struct T5ModelBuilder {
+    device: Device,
+    config: t5::Config,
+    weights_filename: PathBuf,
+}
+
+impl T5ModelBuilder {
```
Comment (EricLBuehler) [inline tests/test_flan-t5-quantized.rs:87]: Would become an impl of ModelLoader?

File: tests/test_flan-t5-quantized.rs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    fn get_local_or_remote_file(filename: &str, api: &ApiRepo) -> Result<PathBuf> {
+        let local_filename = std::path::PathBuf::from(filename);
+        if local_filename.exists() {
+            Ok(local_filename)
+        } else {
+            Ok(api.get(filename)?)
+        }
+    }
+}
+fn generate_answer(_prompt: String, args: &Args) -> Result<String> {
```
Comment (EricLBuehler) [inline tests/test_flan-t5-quantized.rs:153]: Could be part of the `ModulePipeline` impl, would need to be modified in order to support streaming.

File: tests/test_flan-t5-quantized.rs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    let mut generated_text = String::new();
+
+    let (_builder, mut _tokenizer) = T5ModelBuilder::load(&args)?;
```
Comment (EricLBuehler) [inline tests/test_flan-t5-quantized.rs:157]: Clippy error: unnecessary reference

**Review by EricLBuehler — CHANGES_REQUESTED** [review]
Looks good, but if you could fix the Clippy error that would be great!

**Comment by EricLBuehler** [comment]
Hi @bm777, thanks for the PR! It looks like you implemented a basic sort of pipeline, please correct me if I am wrong. Otherwise, this looks good and the only thing I can see is the Clippy error.

File: tests/test_flan-t5-quantized.rs (hunk 3 above)
Comment (bm777) [reply tests/test_flan-t5-quantized.rs]: @EricLBuehler done

**Comment by bm777** [comment]
Heya @EricLBuehler, solved the Clippy error. Thanks for your quick reply :)

**Comment by EricLBuehler** [comment]
Thanks!

**Pull request merged**